cfgmgr/portmgr.cpp

```diff
@@ -157,31 +121,57 @@ void PortMgr::doTask(Consumer &consumer)
                 {
                     tpid = fvValue(i);
                 }
+                else 
+                {
+                    field_values.emplace_back(i);
+                }
             }
 
             if (!mtu.empty())
             {
-                setPortMtu(alias, mtu);
+                if (portOk)
+                {
+                    setPortMtu(alias, mtu);
+                }
+                else
+                {
+                    writeConfigToAppDb(alias, "mtu", mtu);
+                    m_retryFields.emplace_back("mtu", mtu);
+                }
                 SWSS_LOG_NOTICE("Configure %s MTU to %s", alias.c_str(), mtu.c_str());
             }
 
             if (!admin_status.empty())
             {
-                setPortAdminStatus(alias, admin_status == "up");
+                if (portOk)
+                {
+                    setPortAdminStatus(alias, admin_status == "up");
+                }
+                else
+                {
+                     writeConfigToAppDb(alias, "admin_status", admin_status);
+                     m_retryFields.emplace_back("admin_status", admin_status);
+                }
                 SWSS_LOG_NOTICE("Configure %s admin status to %s", alias.c_str(), admin_status.c_str());
             }
 
             if (!learn_mode.empty())
             {
-                setPortLearnMode(alias, learn_mode);
+                writeConfigToAppDb(alias, "learn_mode", learn_mode);
                 SWSS_LOG_NOTICE("Configure %s MAC learn mode to %s", alias.c_str(), learn_mode.c_str());
             }
 
             if (!tpid.empty())
             {
-                setPortTpid(alias, tpid);
+                writeConfigToAppDb(alias, "tpid", tpid);
                 SWSS_LOG_NOTICE("Configure %s TPID to %s", alias.c_str(), tpid.c_str());
             }
+
+            for (auto &entry : field_values)
+            {
+                writeConfigToAppDb(alias, fvField(entry), fvValue(entry));
+                SWSS_LOG_NOTICE("Configure %s %s to %s", alias.c_str(), fvField(entry).c_str(), fvValue(entry).c_str());
```

Comment:
i'm confused by this. It seems like we are anyways writing all the fields to APP DB, so why individual writes above like mtu, tpid?

Reply:
I will fix this